File: backend/src/main/java/csc13001/plantpos/domain/models/Customer.java
package csc13001.plantpos.domain.models;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "customers")
public class Customer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "customer_id")
    private Long customerId;

    @Column(name = "name", length = 256)
    private String name;

    @NotBlank(message = "Phone number is required")
    @Column(name = "phone", length = 20, unique = true)
    private String phone;

    @Column(name = "email", length = 256)
    private String email;

    @Column(name = "address")
    private String address;

    @Column(name = "loyalty_points")
    private int loyaltyPoints;

    @Column(name = "loyalty_card_type")
    private String loyaltyCardType;
}

File: csc13001-plant-pos-frontend.Core/Models/Customer.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace csc13001_plant_pos_frontend.Core.Models;

[Table("customers")]
public class Customer
{
    [Key]
    [Column("customer_id")]
    public int CustomerId
    {
        get; set;
    }

    [Column("name")]
    [StringLength(256)]
    public string Name
    {
        get; set;
    }

    [Column("phone")]
    [StringLength(20)]
    public string Phone
    {
        get; set;
    }

    [Column("email")]
    [StringLength(256)]
    public string Email
    {
        get; set;
    }

    [Column("address")]
    public string Address
    {
        get; set;
    }

    [Column("loyalty_points")]
    public int LoyaltyPoints
    {
        get; set;
    }

    [Column("loyalty_card_type")]
    public string LoyaltyCardType
    {
        get; set;
    }
}
